libs/vkd3d/vkd3d_private.h

```diff
@@ -838,6 +839,19 @@ static inline VkImageLayout d3d12_resource_pick_layout(const struct d3d12_resour
     return resource->flags & VKD3D_RESOURCE_LINEAR_TILING ? VK_IMAGE_LAYOUT_GENERAL : layout;
 }
 
+static inline VkImageLayout d3d12_resource_get_outside_render_pass_color_layout(const struct d3d12_resource *resource)
+{
+    if (resource->desc.Flags & D3D12_RESOURCE_FLAG_ALLOW_SIMULTANEOUS_ACCESS)
+        return resource->common_layout;
+    else
+        return d3d12_resource_pick_layout(resource, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL);
+}
+
```

Comment:
wouldn't it be more useful to fold this into `d3d12_resource_pick_layout` instead?

i.e. adding a new resource flag (`VKD3D_RESOURCE_SIMULTANEOUS_ACCESS`) and then changing that to something like
```c
    return resource->flags & (VKD3D_RESOURCE_LINEAR_TILING | VKD3D_RESOURCE_SIMULTANEOUS_ACCESS)
            ? resource->common_layout : layout;
```
This was the intended purpose at least.

Reply:
I think that approach is fine.